repos/system_upgrade/common/actors/setuptargetrepos/libraries/setuptargetrepos_repomap.py

```diff
@@ -0,0 +1,182 @@
+from collections import default_dict
+
+from leapp.libraries.common import config
+from leapp.libraries.common.config.version import get_source_major_version, get_target_major_version
+
+
+DEFAULT_SRC_PESID = {
+    '7': 'rhel7-base',
+    '8': 'rhel8-BaseOS'
+}
+
+
+def _get_channel_prio(pesid_repo):
+    priorities = {
+        'beta': 0,
+        'ga': 1,
+    }
+    return priorities.get(pesid_repo.channel, 10)
+
+
+class RepoMapDataHandler(object):
+    """
+    Provide the basic functionality to work with the repository data easily.
+    """
+
+    def __init__(self, repo_map, default_channels=None):
+        """
+        Initialize the object based on the given RepositoriesMapping msg.
+
+        Expects that msg contains just stuff related for the current IPU
+        (at least mapping and repos for the used upgrade path and architecture).
+        """
+        # NOTE: currently I am keeping this default data structure that is not
+        # ideal for work, but there is not any significant impact..
+        self.repositories = repo_map.repositories
+        self.mapping = repo_map.mapping
+        self.default_channels = default_channels or ['ga']
+
+    def set_default_channels(self, default_channels):
+        """
+        Set the default channels that are used as a fallback when searching
+        for the right target repository.
+
+        Usually it's not problem to find a target repository that matches
+        the source repository, however in some cases the target repository
+        doesn't have to be available in the required (premium) channel but
+        could be present in the standard one.
+
+        This is used usually for fallbacks and it's prerequisite for the time
+        the required channel will not be present for the particular repository.
+        E.g. can happen for layered products which has different lifecycles
+        and doesn't have to provide a special premium channels at all. E.g.
+        the Extras repository has GA and Beta channels, but no EUS. In case the
+        EUS is required, returns the GA one instead when this is present.
+
+        It's recommended to make the GA ('ga') always present in the default
+        list.
+        """
+        self.default_channels = default_channels
+
+    def get_pesid_repo_entry(self, repoid, major_version):
+        """
+        Return the PESIDRepositoryEntry based on the given repoid and OS major version.
+        """
+        for pesid_repo in self.repositories:
+            if pesid_repo.repoid == repoid and pesid_repo.major_version == major_version:
+                return pesid_repo
+        return None
+
+    def get_target_pesids(self, src_pesid):
+        """
+        Return the list of target PES IDs for the given source PES ID.
+        """
+        pesids = []
+        for repomap in self.mapping:
+            if repomap.source == src_pesid:
+                pesids += pesid
```

Comment:
Not sure whether I see where `pesid` is defined.

Reply:
it wasn't :) it's fixed already.